instrumentation/opentelemetry-instrumentation-django/src/opentelemetry/instrumentation/django/middleware.py

```diff
@@ -146,15 +152,27 @@ def process_request(self, request):
 
         request_meta = request.META
 
-        token = attach(extract(request_meta, getter=wsgi_getter))
+        context = extract(request_meta, getter=wsgi_getter)
+
+        token = attach(context)
 
-        span = self._tracer.start_span(
-            self._get_span_name(request),
-            kind=SpanKind.SERVER,
-            start_time=request_meta.get(
-                "opentelemetry-instrumentor-django.starttime_key"
-            ),
-        )
+        if get_current_span() is INVALID_SPAN:
+            span = self._tracer.start_span(
+                self._get_span_name(request),
+                context,
+                kind=SpanKind.SERVER,
+                start_time=request_meta.get(
+                    "opentelemetry-instrumentor-django.starttime_key"
+                ),
+            )
+        else:
+            span = self._tracer.start_span(
+                self._get_span_name(request),
+                kind=SpanKind.INTERNAL,
+                start_time=request_meta.get(
+                    "opentelemetry-instrumentor-django.starttime_key"
+                ),
```

Comment:
blocks like these could be turned into a reusable function

Reply:
I did not end up turning these blocks into a helper method; as I mentioned in an earlier comment, different instrumentations require different attributes and have separate logic for extracting and activating context. I thought my helper method would be unnecessarily complex if I took into account all these differences, so I just implemented the logic directly in each instrumentation.